source/PlayerInfo.cpp

```diff
@@ -2794,6 +2794,19 @@ void PlayerInfo::Save(const string &path) const
 		out.Write("travel", system->Name());
 	if(travelDestination)
 		out.Write("travel destination", travelDestination->TrueName());
+	// Detect which ship number is the current flagship, for showing on LoadPanel
+	if(flagship)
+	{
+		for(auto it = ships.begin(); it != ships.end(); ++it)
+		{
+			if(*it == flagship)
+				out.Write("flagship index", distance(ships.begin(), it));
+		}
+	}
+	else
+	{
+		out.Write("flagship index", -1);
+	}
```

Comment:
Remove unnecessary braces. Also, break out of the loop once we have the flagship index
```suggestion
	if(flagship)
	{
		for(auto it = ships.begin(); it != ships.end(); ++it)
			if(*it == flagship)
			{
				out.Write("flagship index", distance(ships.begin(), it));
				break;
			}
	}
	else
		out.Write("flagship index", -1);
```